packages/ui/src/hooks/useStyles.ts

```diff
@@ -66,27 +151,10 @@ export function useStyles<T extends ComponentDefinitionName>(
   dataAttributes: Record<string, string>;
   resolvedProps: Record<string, string>;
 } {
-  const { breakpoint } = useBreakpoint();
-  const classNames = componentDefinitions[componentName]
-    .classNames as ComponentClassNames<T>;
-
-  // Resolve responsive values and generate data attributes
-  const dataAttributes: Record<string, string> = {};
-  const resolvedProps: Record<string, string> = {};
-
-  for (const [key, value] of Object.entries(props)) {
-    if (value !== undefined && value !== null) {
-      const resolvedValue = resolveResponsiveValue(value, breakpoint);
-      if (resolvedValue !== undefined) {
-        resolvedProps[key] = resolvedValue;
-        dataAttributes[`data-${key}`] = resolvedValue;
-      }
-    }
+  // Choose the appropriate hook based on whether props contain responsive values
+  if (hasResponsiveProps(props)) {
+    return useStylesResponsive(componentName, props);
+  } else {
+    return useStylesNonResponsive(componentName, props);
```

Comment:
Hmm, technically this is not allowed, because you're not guaranteed that the props stay the same. I'd say the risk of that is big enough that we can't go with this approach. I wonder if there's something we can do to make `useBreakpoint` more lazy instead for example?